shipctl/x86_64/WindowsServer_2016/utility.ps1

```diff
@@ -728,3 +786,106 @@ function _send_notification() {
     }
   }
 }
+
+function _send_irc_notification() {
+  param(
+    [string]$server,
+    [string]$channel,
+    [string]$nick,
+    [string]$password,
+    [string]$payload
+  )
+  $port = 6667
+
+  # make the connection
+  $client = New-Object -TypeName Net.Sockets.TcpClient
+  $client.Connect($server, $port)
+  $netStream = $client.GetStream()
+  $encoding = [System.Text.Encoding]::ASCII
+  $streamWriter = New-Object -Type System.IO.StreamWriter -ArgumentList $netStream, $encoding
+  $connection = "on"
+  # send initial login commands
+  $streamWriter.WriteLine("PASS $password")
+  $streamWriter.Flush()
+  $streamWriter.WriteLine("NICK $nick")
+  $streamWriter.WriteLine("USER $env:UserName 0 * :Shippable Assembly Lines")
+  $streamWriter.Flush()
+
+  # read data
+  $maxWaitTime = 60 # seconds
+  $message = ""
+  $joined = $false
+  while (($connection -eq "on") -and ($maxWaitTime -ge 0)) {
+
+    if ($netStream.DataAvailable) {
+      [char]$char = $netStream.ReadByte()
+      if ($char -eq 13) {
+        write-output "$message"
+        $joined = _check_message $streamWriter $message $channel
+        if ($joined) {
+        }
+        $message = ""
+      } elseif ($char -ne 10) {
+        $message += $char
+      }
+    } else {
+      if ($joined -eq $true) {
+        # send payload
+        $streamWriter.WriteLine("JOIN $channel")
+        $streamWriter.Flush()
+        $payload = $payload.TrimEnd()
+        $streamWriter.WriteLine("NOTICE $channel :$payload")
+        $streamWriter.Flush()
+        sleep 2
+        $connection = "off"
+      } else {
+        write-output "waiting for $maxWaitTime more seconds..."
+        sleep 2
+        $maxWaitTime -= 2
+      }
+    }
+  }
+  if ($maxWaitTime -le 0) {
+    Write-Output "Max wait time exceeded. Closing connection."
+    $errorMessage = "Max wait time exceeded"
+  }
+
+  # disconnect
+  write-output "disconnecting from server"
+  $streamWriter.WriteLine("PART $channel")
+  $streamWriter.WriteLine("QUIT")
+  $streamWriter.Flush()
+  $streamWriter.Close()
+  $netStream.Close()
+  $client.Close()
+  if ($errorMessage) {
+    throw $errorMessage
+  }
+}
+
+function _check_message() {
+
+  param(
+    [System.IO.StreamWriter]$writer,
+    [string]$message,
+    [string]$channel
+  )
+
+  if ($message -match "^:(.+?) +([0-9]{3}|[A-Z]+)") {
+    $prefix = $matches[1]
+    $code = $matches[2]
+  }
+
+  $joined = $false
+  switch -regex ($code) {
+    "376" { # means end of MOTD. further commands will be accepted now
+      write-output "joining channel $channel"
+      # received message of the day
+      $writer.WriteLine("JOIN $channel")
+      $writer.Flush()
+      $joined = $true
```

Comment:
Do you need to wait for confirmation that the JOIN worked?

Reply:
it seems not necessary since the payload sends, and i couldn't tell what the confirmation would look like, but i will investigate this more today.